superset-frontend/src/reports/reducers/reports.js

```diff
@@ -94,7 +53,7 @@ export default function reportsReducer(state = {}, action) {
           ...state,
           charts: {
             ...state.chart,
-            [report.id]: report,
+            [result.chart]: report,
```

Comment:
can we do the same thing with these actions where we dry up some of the code to look up the chart or dashboard like above? Maybe in the next PR.